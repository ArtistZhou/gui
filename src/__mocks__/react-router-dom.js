import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
