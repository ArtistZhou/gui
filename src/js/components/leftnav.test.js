import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import LeftNav from './leftnav';

const mockStore = configureStore([]);
const store = mockStore({
  app: { features: { isHosted: false }, versionInformation: {} }
});
it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <LeftNav />
        </MemoryRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
