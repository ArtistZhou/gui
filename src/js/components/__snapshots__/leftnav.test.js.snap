// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <ul
    className="MuiList-root-1 MuiList-padding-2"
    style={
      Object {
        "padding": "0",
      }
    }
  >
    <a
      aria-current="page"
      className="MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-13 navLink leftNav active"
      disabled={false}
      href="/"
      onClick={[Function]}
      style={
        Object {
          "padding": "22px 16px 22px 42px",
        }
      }
    >
      <div
        className="MuiListItemText-root-17"
        style={
          Object {
            "textTransform": "uppercase",
          }
        }
      >
        <span
          className="MuiTypography-root-23 MuiTypography-subheading-30 MuiListItemText-primary-20"
        >
          Dashboard
        </span>
      </div>
    </a>
    <a
      aria-current={null}
      className="MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-13 navLink leftNav"
      disabled={false}
      href="/devices"
      onClick={[Function]}
      style={
        Object {
          "padding": "22px 16px 22px 42px",
        }
      }
    >
      <div
        className="MuiListItemText-root-17"
        style={
          Object {
            "textTransform": "uppercase",
          }
        }
      >
        <span
          className="MuiTypography-root-23 MuiTypography-subheading-30 MuiListItemText-primary-20"
        >
          Devices
        </span>
      </div>
    </a>
    <a
      aria-current={null}
      className="MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-13 navLink leftNav"
      disabled={false}
      href="/releases"
      onClick={[Function]}
      style={
        Object {
          "padding": "22px 16px 22px 42px",
        }
      }
    >
      <div
        className="MuiListItemText-root-17"
        style={
          Object {
            "textTransform": "uppercase",
          }
        }
      >
        <span
          className="MuiTypography-root-23 MuiTypography-subheading-30 MuiListItemText-primary-20"
        >
          Releases
        </span>
      </div>
    </a>
    <a
      aria-current={null}
      className="MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-13 navLink leftNav"
      disabled={false}
      href="/deployments"
      onClick={[Function]}
      style={
        Object {
          "padding": "22px 16px 22px 42px",
        }
      }
    >
      <div
        className="MuiListItemText-root-17"
        style={
          Object {
            "textTransform": "uppercase",
          }
        }
      >
        <span
          className="MuiTypography-root-23 MuiTypography-subheading-30 MuiListItemText-primary-20"
        >
          Deployments
        </span>
      </div>
    </a>
  </ul>
  <ul
    className="MuiList-root-1 MuiList-padding-2"
    style={
      Object {
        "bottom": "30px",
        "left": "0px",
        "padding": "0",
        "position": "absolute",
        "right": "0px",
      }
    }
  >
    <a
      className="MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-13 navLink leftNav"
      disabled={false}
      href="/help"
      onClick={[Function]}
      style={
        Object {
          "padding": "16px 16px 16px 42px",
        }
      }
    >
      <div
        className="MuiListItemText-root-17"
      >
        <span
          className="MuiTypography-root-23 MuiTypography-subheading-30 MuiListItemText-primary-20"
        >
          Help
        </span>
      </div>
    </a>
    <li
      className="MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-13 MuiListItem-disabled-11"
      disabled={true}
      style={
        Object {
          "color": "#949495",
          "padding": "16px 16px 16px 42px",
        }
      }
    >
      <div
        className="MuiListItemText-root-17"
      >
        <span
          className="MuiTypography-root-23 MuiTypography-subheading-30 MuiListItemText-primary-20"
        >
          <div
            aria-describedby={null}
            className=""
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            title={null}
          >
            
          </div>
        </span>
        <p
          className="MuiTypography-root-23 MuiTypography-body1-32 MuiTypography-colorTextSecondary-56 MuiListItemText-secondary-21"
        >
          <a
            href="https://docs.mender.io/development/release-information/open-source-licenses"
            rel="noopener noreferrer"
            style={
              Object {
                "color": "#347A87",
                "fontSize": "13px",
                "position": "relative",
                "top": "6px",
              }
            }
            target="_blank"
          >
            License information
          </a>
        </p>
      </div>
    </li>
  </ul>
</div>
`;
